client/state/utils.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import tv4 from 'tv4';
+
+/**
+ * Internal dependencies
+ */
+import warn from 'lib/warn';
+
+/**
+ * Module variables
+ */
+
+export function isValidStateWithSchema( state, schema, checkForCycles = false, banUnknownProperties = false ) {
+	const result = tv4.validateResult( state, schema, checkForCycles, banUnknownProperties );
```

Comment:
banUnknown wins, this is generally only useful if you're debugging something. I added checkForCycles, and bandUnknownProperties to the interface so I wouldn't have to look up tv4 documentation later for the extra bools. I can remove this if its distracting.